civihr_employee_portal/src/Webform/LocationTypeIDConvertor.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\Webform;
+
+class LocationTypeIDConvertor implements WebformTransferConvertor {
+
+  const MAPPING_KEY = 'locationTypes';
+
+  /**
+   * @inheritdoc
+   */
+  public static function preExport(\stdClass $node) {
+    $locationTypeIDs = self::getLocationTypeIDsByRef($node);
+    $params = ['id' => ['IN' => $locationTypeIDs]];
+    $locationTypes = civicrm_api3('LocationType', 'get', $params)['values'];
+    $locationTypes = array_column($locationTypes, 'name', 'id');
+    $node->customMapping[self::MAPPING_KEY] = $locationTypes;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public static function preImport(\stdClass $node) {
+    $oldMapping = $node->customMapping[self::MAPPING_KEY];
+    $oldToNewMapping = self::reverseMapping($oldMapping);
+
+    foreach (self::getLocationTypeIDsByRef($node) as &$locationTypeID) {
+      if (isset($oldToNewMapping[$locationTypeID])) {
+        $locationTypeID = $oldToNewMapping[$locationTypeID];
+      }
+    }
+  }
+
+  /**
+   * Gets all location type IDs for a webform.
+   * Results are returned by reference to allow easier modification.
+   *
+   * @param \stdClass $node
+   * @return array
+   */
+  private static function getLocationTypeIDsByRef(\stdClass $node) {
+    $locationTypeIDs = [];
+
+    $civicrmData = &$node->webform_civicrm;
```

Comment:
Do we need to pass most of the variables in this function by reference? Since we are not modifying the variables directly but just adding to the `$locationTypeIDs` array, I don't think this is necessary.
Or what do you think?

Reply:
This isn't required for the first call to this, but when actually changing the values from `preImport` passing by reference saves repetition (since the same method can be used for just fetching the values) and avoids having to directly change the values like

`$node->civicrm_webform['data']['Contact'][0]['phone'][0]['location_type_id'] = $newID`